src/App.jsx

```diff
@@ -1088,12 +1088,38 @@ const App = memo(() => {
     if (!playing) {
       video.play();
     } else {
+      let calculatedRate;
+      if (useMultiplier) {
+        switch (true) {
+          case speedMultiplier === '2x' && dir > 0:
+            calculatedRate = video.playbackRate * 2.0;
+            break;
+          case speedMultiplier === '2x' && dir < 0:
+            calculatedRate = video.playbackRate * 0.5;
+            break;
+          case speedMultiplier === '1.1x' && dir > 0:
+            calculatedRate = video.playbackRate * 1.1;
+            break;
+          case speedMultiplier === '1.1x' && dir < 0:
+            calculatedRate = video.playbackRate * 0.9091;
+            break;
+          default:
+            calculatedRate = 1.0;
+            break;
+        }
+      } else {
+        calculatedRate = video.playbackRate + (dir * 0.15);
+      }
+      // if the multiplier causes us to go faster than real time or slower than real time, stop along the way at 1.0
+      if ((calculatedRate > 1.0 && video.playbackRate < 1.0) || (calculatedRate < 1.0 && video.playbackRate > 1.0)) {
+        calculatedRate = 1.0;
+      }
```

Comment:
This logic risks skipping a step. Consider what happens when rounding errors cause the rate, while stepping down, to be sliiiightly above 1.0. Now you step down and end up at 1.0.

Better: set the rate to 1.0 if it is between multiplier^(-1/2) and multiplier^(1/2).

Reply:
Oh you mean that if playbackRate is high and the user is pressing J-J-J, they may end up at 1.00001 which shows up in the UI as 100%, and then they press J again, and end up at 1, which also shows as 100%?

Reply:
Exactly.